src/app/qgisapp.cpp

```diff
@@ -5040,6 +5189,12 @@ bool QgisApp::addVectorLayersPrivate( const QStringList &layerQStringList, const
   QgsProject::instance()->addMapLayers( layersToAdd );
   for ( QgsMapLayer *l : qgis::as_const( layersToAdd ) )
   {
+    // Connect style changed signal to run dependencies checks
+    QgsVectorLayer *vl { qobject_cast< QgsVectorLayer * >( l ) };
```

Comment:
Instead of implementing this here, I think it would be safer to listen to the project's layer added signals and respond to that. There's many ways to add layers to qgis which don't go through addVectorLayersPrivate

Reply:
Do you mean the connection, right?